app/project_type/git.py

```diff
@@ -149,25 +150,38 @@ def _fetch_project(self):
 
         # Must add the --update-head-ok in the scenario that the current branch of the working directory
         # is equal to self._branch, otherwise the git fetch will exit with a non-zero exit code.
-        # Must specify the colon in 'branch:branch' so that the branch will be created locally. This is
-        # important because it allows the slave hosts to do a git fetch from the master for this branch.
-        fetch_command = 'git fetch --update-head-ok {0} {1}:{1}'.format(self._remote, self._branch)
+        fetch_command = 'git fetch --update-head-ok {} {}'.format(self._remote, self._branch)
```

Comment:
Yeah, we really don't need to create the branch locally as long as we have the FETCH_HEAD. +1.

I'm guessing this line also gets rid of the possibility of throwing that fast-forward error when you do a "git fetch scm REMOTE_BRANCH:LOCAL_BRANCH"  in the scenario where the remote branch has somehow gotten behind the local branch.